### DISCUSSION TITLE
how can I make the carousel start at a certain slide?

### DISCUSSION THREAD

**Discussion by asdrubalp9** [quasarframework/quasar#11645] [General - Components / Directives / etc] [answered]
I have a modal in a component and I parse an object and one of the items of the object is an array with images,  and sometimes the array has one or many images.

So if I open the modal that has many images and slide it to the third image, then close the modal to open one with one image, the carousel has the key at the third position of the image array.

so how can I make this start at the first position of the array? 

`
<q-carousel

                    v-model="slide"

                    transition-prev="slide-right"

                    transition-next="slide-left"

                    swipeable

                    animated

                    control-color="primary"

                    navigation

                    padding

                    arrows

                    height="300px"

                    class="rounded-borders"

                    v-if="hasImages"

                    infinite

                    :autoplay="true"
                >

                    <q-carousel-slide :name="k"  class="column no-wrap" v-for="(img, k) in order.imgs" :key="k" >

                        <div class="row fit justify-center items-center q-gutter-xs q-col-gutter no-wrap">

                            <q-img class="rounded-borders col-6 full-height" :src="`http://domain.test/${img}`" />

                        </div>
                    </q-carousel-slide>

                </q-carousel>
`

**Comment by MilosPaunovic** [comment, marked as answer]
Set `slide` variable used for `QCarousel`s `v-model` to the `name` value of slide you want to show.

**Reply by asdrubalp9** [reply]
Thanks! 
Although it wasn't the answer, it was a very good clue.

I fixed it by setting the slide variable to 0 when the modal closed.

**Reply by MilosPaunovic** [reply]
Well, the value of `k` (which is `index` in your `v-for` and also `name` property of your `QCarouselSlide`) is zero for the first image, so that is the reason why setting the `slide` variable to zero is showing the first slide.
